asm/f64/axpy_test.go

```diff
@@ -6,533 +6,262 @@ package f64
+	},
+	{
+		alpha:   1,
+		x:       []float64{2, 3, 4},
+		y:       []float64{-3, -2, -1},
+		want:    []float64{-1, 1, 3},
+		wantRev: []float64{1, 1, 1},
+	},
+	{
 		alpha:   0,
-		xData:   []float64{0, 0, 1, 1, 2, -3, -4},
-		yData:   []float64{0, 1, 0, 3, -4, 5, -6},
+		x:       []float64{0, 0, 1, 1, 2, -3, -4},
+		y:       []float64{0, 1, 0, 3, -4, 5, -6},
 		want:    []float64{0, 1, 0, 3, -4, 5, -6},
 		wantRev: []float64{0, 1, 0, 3, -4, 5, -6},
 	},
 	{
 		alpha:   1,
-		xData:   []float64{0, 0, 1, 1, 2, -3, -4},
-		yData:   []float64{0, 1, 0, 3, -4, 5, -6},
+		x:       []float64{0, 0, 1, 1, 2, -3, -4},
+		y:       []float64{0, 1, 0, 3, -4, 5, -6},
 		want:    []float64{0, 1, 1, 4, -2, 2, -10},
 		wantRev: []float64{-4, -2, 2, 4, -3, 5, -6},
 	},
 	{
 		alpha:   3,
-		xData:   []float64{0, 0, 1, 1, 2, -3, -4},
-		yData:   []float64{0, 1, 0, 3, -4, 5, -6},
+		x:       []float64{0, 0, 1, 1, 2, -3, -4},
+		y:       []float64{0, 1, 0, 3, -4, 5, -6},
 		want:    []float64{0, 1, 3, 6, 2, -4, -18},
 		wantRev: []float64{-12, -8, 6, 6, -1, 5, -6},
 	},
 	{
 		alpha:   -3,
-		xData:   []float64{0, 0, 1, 1, 2, -3, -4},
-		yData:   []float64{0, 1, 0, 3, -4, 5, -6},
-		want:    []float64{0, 1, -3, 0, -10, 14, 6},
-		wantRev: []float64{12, 10, -6, 0, -7, 5, -6},
+		x:       []float64{0, 0, 1, 1, 2, -3, -4, 0, 0, 1, 1, 2, -3, -4},
+		y:       []float64{0, 1, 0, 3, -4, 5, -6, 0, 1, 0, 3, -4, 5, -6},
+		want:    []float64{0, 1, -3, 0, -10, 14, 6, 0, 1, -3, 0, -10, 14, 6},
+		wantRev: []float64{12, 10, -6, 0, -7, 5, -6, 12, 10, -6, 0, -7, 5, -6},
 	},
 	{
 		alpha:   -5,
-		xData:   []float64{0, 0, 1, 1, 2, -3, -4, 5},
-		yData:   []float64{0, 1, 0, 3, -4, 5, -6, 7},
-		want:    []float64{0, 1, -5, -2, -14, 20, 14, -18},
-		wantRev: []float64{-25, 21, 15, -7, -9, 0, -6, 7},
+		x:       []float64{0, 0, 1, 1, 2, -3, -4, 5, 1, 2, -3, -4, 5},
+		y:       []float64{0, 1, 0, 3, -4, 5, -6, 7, 3, -4, 5, -6, 7},
+		want:    []float64{0, 1, -5, -2, -14, 20, 14, -18, -2, -14, 20, 14, -18},
+		wantRev: []float64{-25, 21, 15, -7, -9, -20, 14, 22, -7, -9, 0, -6, 7},
 	},
 }
 
-func TestDaxpyUnitary(t *testing.T) {
-	for i, test := range daxpyTests {
-		const msgGuard = "%v: out-of-bounds write to %v argument\nfront guard: %v\nback guard: %v"
-
-		prefix := fmt.Sprintf("test %v (y+=a*x)", i)
-		x, xFront, xBack := newGuardedVector(test.xData, 1)
-		y, yFront, yBack := newGuardedVector(test.yData, 1)
-		AxpyUnitary(test.alpha, x, y)
-
-		if !allNaN(xFront) || !allNaN(xBack) {
-			t.Errorf(msgGuard, prefix, "x", xFront, xBack)
-		}
-		if !allNaN(yFront) || !allNaN(yBack) {
-			t.Errorf(msgGuard, prefix, "y", yFront, yBack)
-		}
-		if !equalStrided(test.xData, x, 1) {
-			t.Errorf("%v: modified read-only x argument", prefix)
-		}
-
-		if !equalStrided(test.want, y, 1) {
-			t.Errorf("%v: unexpected result:\nwant: %v\ngot: %v", prefix, test.want, y)
+func TestAxpyUnitary(t *testing.T) {
+	var xGdVal, yGdVal float64 = -1, 0.5
+	for i, tt := range axpyTests {
```

Comment:
s/tt/test/g; in tests `test.foo` is a convention here for referring to test records.